### PR TITLE
Improve the "TSCN file format" page

### PR DESCRIPTION
**Pull request by Calinou**
This also mentions the ability to write single-line comments in TSCN/TRES files.

### CODE REVIEW

File: development/file_formats/tscn.rst [hunk 1]
```diff
@@ -1,84 +1,87 @@
 TSCN file format
 ================
 
-A :code:`.tscn` File format is the "Text SCeNe" file format and represents
-a single scene-tree inside Godot. TSCN files have the advantage of being
-nearly human-readable and easy for version control systems to manage. During
-import the TSCN files are compiled into binary :code:`.scn` files stored
-inside the .import folder. This reduces the data size and speed up loading.
+The TSCN (text scene) file represents a single scene tree inside
```
Comment (aaronfranke) [inline development/file_formats/tscn.rst]: Either "a TSCN file" or "the TSCN file format", saying "the TSCN file" implies only one file.

File: development/file_formats/tscn.rst [hunk 2]
```diff
@@ -1,84 +1,87 @@
...
+Godot. TSCN files have the advantage of being mostly human-readable and easy for
+version control systems to manage. During import, TSCN files are compiled into
+binary ``.scn`` files stored inside the .import folder. This reduces the data
+size and speeds up loading.
 
-The :code:`.escn` file format is identical to the TSCN file format, but is used to
+The ``.escn`` file format is identical to the TSCN file format, but is used to
```
Comment (aaronfranke) [inline development/file_formats/tscn.rst]: Why `.escn` here but `TSCN` above?

File: development/file_formats/tscn.rst [hunk 3]
```diff
@@ -1,84 +1,87 @@
...
 4. Connections
 
-The file descriptor looks like :code:`[gd_scene load_steps=1 format=2]` And
-should be the first entry in the file. The load_steps parameter should (in
-theory) be the number of resources within the file, though in practice its
-value seems not to matter.
+The file descriptor looks like ``[gd_scene load_steps=1 format=2]`` and should
+be the first entry in the file. The ``load_steps`` parameter should (in theory)
+be the number of resources within the file. However, in practice, its value seems
+not to matter.
```
Comment (aaronfranke) [inline development/file_formats/tscn.rst]: I know this is existing text, but if the value seems not to matter, then there should be an explanation as to why it's stored at all.

File: development/file_formats/tscn.rst [hunk 4]
```diff
@@ -1,84 +1,87 @@
...
 
-The first node in the file should not have the :code:`parent=Path/To/Node`
-entry in it's heading, and it is the scene root. All scene files should have
-exactly one scene root. It it does not, Godot will fail to import the file.
-The parent path of other nodes should be absolute, but without the scene
-root's name. If it is a direct child of the scene root, it should be
-:code:`"."`. Here is an example scene tree (but without any node content).
+The first node in the file, which is also the scene root, must not have a
+``parent=Path/To/Node`` entry in its heading. All scene files should have
+exactly *one* scene root. It it doesn't, Godot will fail to import the file. The
```
Comment (aaronfranke) [inline development/file_formats/tscn.rst]: It it -> If it

File: development/file_formats/tscn.rst [hunk 5]
```diff
@@ -1,84 +1,87 @@
...
+parent path of other nodes should be absolute, but shouldn't contain the scene
+root's name. If the node is a direct child of the scene root, it should be ``"."``.
```
Comment (aaronfranke) [inline development/file_formats/tscn.rst]: it -> the path

File: development/file_formats/tscn.rst [hunk 6]
```diff
@@ -299,45 +306,46 @@ under each heading. For example, a capsule collision shape looks like:
...
-        2) Tangents array
-        3) Vertex color array
-        4) UV array 1
-        5) UV array 2
-        6) Bone index array
-        7) Bone weight array
-        8) Vertex index array
+        1. An array of vertex positions
+        2. Tangents array
+        3. Vertex colors array
```
Comment (aaronfranke) [inline development/file_formats/tscn.rst]: An array of vertex positions -> Vertex positions array

File: development/file_formats/tscn.rst (hunk 2 above)
Comment (Calinou) [reply development/file_formats/tscn.rst]: I don't understand what you meant here. Could you clarify, please?

File: development/file_formats/tscn.rst (hunk 3 above)
Comment (Calinou) [reply development/file_formats/tscn.rst]: I'm not sure what it's used for either. I guess it's used to display a progress bar with ResourceInteractiveLoader. However, if its number is incorrect, the resource will still load correctly (but the progress bar may be wrong).

File: development/file_formats/tscn.rst (hunk 2 above)
Comment (aaronfranke) [reply development/file_formats/tscn.rst]: "The TSCN (text scene) file format" vs "The `.escn` file format", one has the format capitalized, the other has a dot and then the format name.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**